webapp/channels/src/packages/mattermost-redux/src/actions/posts.ts

```diff
@@ -230,165 +236,66 @@ export function createPost(post: Post, files: any[] = []): ActionFuncAsync {
 
         dispatch(batchActions(actions, 'BATCH_CREATE_POST_INIT'));
 
-        (async function createPostWrapper() {
-            try {
-                const created = await Client4.createPost({...newPost, create_at: 0});
+        try {
```

Comment:
@hmhealey That makes sense. 

I would prefer 2 as that doesn't depend on waiting for a websocket event and lining up events and makes the interface much easier to understand.

Reply:
@hmhealey Did we test whether the optimistic approach stopped working? Looking into the code, it would seem that most of the "importat stuff" was not being waited upon, so the changes should not break the optimistic posting. But not sure if I properly tested that particular part...